pallets/omnipool/src/lib.rs

```diff
@@ -2017,18 +2124,48 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Calls `on_trade_fee` hook and ensures that no more than the fee amount is transferred.
-	fn process_trade_fee(trader: &T::AccountId, asset: T::AssetId, amount: Balance) -> DispatchResult {
+	fn process_trade_fee(
+		trader: &T::AccountId,
+		asset: T::AssetId,
+		amount: Balance,
+	) -> Result<Vec<Fee<T::AccountId>>, DispatchError> {
 		let account = Self::protocol_account();
 		let original_asset_reserve = T::Currency::free_balance(asset, &account);
 
 		// Let's give little bit less to process. Subtracting one due to potential rounding errors
 		let allowed_amount = amount.saturating_sub(Balance::one());
-		let used = T::OmnipoolHooks::on_trade_fee(account.clone(), trader.clone(), asset, allowed_amount)?;
+		let fees = T::OmnipoolHooks::on_trade_fee(account.clone(), trader.clone(), asset, allowed_amount)?;
+		let additional_fees_total: Balance = fees
```

Comment:
it is not really additional fee. so i would suggest to change naming here. eg "taken_fee"